botocore/client.py

```diff
@@ -239,16 +163,18 @@ def _get_client_args(self, service_model, region_name, is_secure,
             if client_config.user_agent_extra is not None:
                 user_agent += ' %s' % client_config.user_agent_extra
 
-        signer = RequestSigner(service_model.service_name, region_name,
-                               service_model.signing_name,
-                               signature_version, credentials,
-                               event_emitter)
+        signer = RequestSigner(
+            service_name, endpoint_config['signing_region'],
+            endpoint_config['signing_name'],
```

Comment:
Good catch. I just pushed an update that adds the missing credentialScope information for all of the services that are currently called out in the model, and I also added support for taking the `signingName` from models if `credentialScope.service` is not present. I was able to connect and successfully call each of these services with this change, and we will now continue to honor `signingName`.